File: robocode-tankroyale-bot-api-dotnet-core/events/BulletFiredEvent.cs
namespace Robocode.TankRoyale
{
  /// <summary>
  /// Event occurring when a bullet has been fired from a bot.
  /// </summary>
  public class BulletFiredEvent : Event
  {
    /// <summary>Bullet that was fired.<summary>
    BulletState Bullet { get; }

    /// <summary>
    /// Constructor.
    /// </summary>
    /// <param name="turnNumber">Turn number.</param>
    /// <param name="bullet">Bullet that was fired.</param>
    public BulletFiredEvent(int turnNumber, BulletState bullet) : base(turnNumber) => Bullet = bullet;
  }
}

File: bot-api/python/src/robocode_tank_royale/bot_api/events/bullet_fired_event.py
from dataclasses import dataclass

from ..bullet_state import BulletState
from .bot_event import BotEvent


@dataclass(frozen=True, repr=True)
class BulletFiredEvent(BotEvent):
    """
    Represents an event that occurs when a bullet is fired from a bot.

    This event contains information about the bullet that was fired and
    the turn number during which the event occurred.

    Attributes:
        bullet (BulletState): The state of the bullet that was fired.
    """
    bullet: BulletState
